error: this use of "defined" may not be portable [-Werror=expansion-to-defined]

(See also https://github.com/cisco/cjose/commit/8693c22aabf31313a4002838e124e93879bbb50b#commitcomment-27669403)

Hi there,

Since 
> * compile with LibreSSL  ([8693c22aabf31313a4002838e124e93879bbb50b](https://github.com/linuxwolf/cjose/commit/8693c22aabf31313a4002838e124e93879bbb50b))

building with newer GCC (and Clang supposedly) produces:
```
util.c: In function 'cjose_apply_allocs':
util.c:53:36: error: this use of "defined" may not be portable [-Werror=expansion-to-defined]
 #if (CJOSE_OPENSSL_11X)
     ^~~~~~~~~~~~~~~~~
```
(Probably due to this patch: https://gcc.gnu.org/ml/gcc-patches/2016-08/msg00738.html)

Maybe there is a way to restructure this macros.

Regards,
Moritz

--- a/include/cjose/util.h
+++ b/include/cjose/util.h
@@ -24,7 +24,9 @@
 extern "C" {
 #endif
 
-#define CJOSE_OPENSSL_11X OPENSSL_VERSION_NUMBER >= 0x10100005L && !defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER >= 0x10100005L && !defined(LIBRESSL_VERSION_NUMBER)
+#define CJOSE_OPENSSL_11X
+#endif
 
 /**
  * Macro to explicitly mark a parameter unused, and usable across multiple

--- a/src/jwe.c
+++ b/src/jwe.c
@@ -875,7 +875,7 @@ static bool _cjose_jwe_set_iv_aes_cbc(cjose_jwe_t *jwe, cjose_err *err)
     return true;
 }
 
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
 #define CJOSE_EVP_CTRL_GCM_GET_TAG EVP_CTRL_AEAD_GET_TAG
 #define CJOSE_EVP_CTRL_GCM_SET_TAG EVP_CTRL_AEAD_SET_TAG
 #else

--- a/src/jwk.c
+++ b/src/jwk.c
@@ -54,7 +54,7 @@ void _cjose_jwk_rsa_get(RSA *rsa, BIGNUM **rsa_n, BIGNUM **rsa_e, BIGNUM **rsa_d
 {
     if (rsa == NULL)
         return;
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
     RSA_get0_key(rsa, (const BIGNUM **)rsa_n, (const BIGNUM **)rsa_e, (const BIGNUM **)rsa_d);
 #else
     *rsa_n = rsa->n;
@@ -78,7 +78,7 @@ bool _cjose_jwk_rsa_set(RSA *rsa, uint8_t *n, size_t n_len, uint8_t *e, size_t e
     if (d && d_len > 0)
         rsa_d = BN_bin2bn(d, d_len, NULL);
 
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
     return RSA_set0_key(rsa, rsa_n, rsa_e, rsa_d) == 1;
 #else
     rsa->n = rsa_n;
@@ -90,7 +90,7 @@ bool _cjose_jwk_rsa_set(RSA *rsa, uint8_t *n, size_t n_len, uint8_t *e, size_t e
 
 void _cjose_jwk_rsa_get_factors(RSA *rsa, BIGNUM **p, BIGNUM **q)
 {
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
     RSA_get0_factors(rsa, (const BIGNUM **)p, (const BIGNUM **)q);
 #else
     *p = rsa->p;
@@ -107,7 +107,7 @@ void _cjose_jwk_rsa_set_factors(RSA *rsa, uint8_t *p, size_t p_len, uint8_t *q, 
     if (q && q_len > 0)
         rsa_q = BN_bin2bn(q, q_len, NULL);
 
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
     RSA_set0_factors(rsa, rsa_p, rsa_q);
 #else
     rsa->p = rsa_p;
@@ -117,7 +117,7 @@ void _cjose_jwk_rsa_set_factors(RSA *rsa, uint8_t *p, size_t p_len, uint8_t *q, 
 
 void _cjose_jwk_rsa_get_crt(RSA *rsa, BIGNUM **dmp1, BIGNUM **dmq1, BIGNUM **iqmp)
 {
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
     RSA_get0_crt_params(rsa, (const BIGNUM **)dmp1, (const BIGNUM **)dmq1, (const BIGNUM **)iqmp);
 #else
     *dmp1 = rsa->dmp1;
@@ -138,7 +138,7 @@ void _cjose_jwk_rsa_set_crt(
     if (iqmp && iqmp_len > 0)
         rsa_iqmp = BN_bin2bn(iqmp, iqmp_len, NULL);
 
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
     RSA_set0_crt_params(rsa, rsa_dmp1, rsa_dmq1, rsa_iqmp);
 #else
     rsa->dmp1 = rsa_dmp1;

--- a/src/jws.c
+++ b/src/jws.c
@@ -268,7 +268,7 @@ static bool _cjose_jws_build_dig_hmac_sha(cjose_jws_t *jws, const cjose_jwk_t *j
     }
 
 // instantiate and initialize a new mac digest context
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
     ctx = HMAC_CTX_new();
 #else
     ctx = cjose_get_alloc()(sizeof(HMAC_CTX));
@@ -279,7 +279,7 @@ static bool _cjose_jws_build_dig_hmac_sha(cjose_jws_t *jws, const cjose_jwk_t *j
         goto _cjose_jws_build_dig_hmac_sha_cleanup;
     }
 
-#if !(CJOSE_OPENSSL_11X)
+#if !defined(CJOSE_OPENSSL_11X)
     HMAC_CTX_init(ctx);
 #endif
 
@@ -316,7 +316,7 @@ static bool _cjose_jws_build_dig_hmac_sha(cjose_jws_t *jws, const cjose_jwk_t *j
 _cjose_jws_build_dig_hmac_sha_cleanup:
     if (NULL != ctx)
     {
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
         HMAC_CTX_free(ctx);
 #else
         HMAC_CTX_cleanup(ctx);
@@ -565,7 +565,7 @@ static bool _cjose_jws_build_sig_ec(cjose_jws_t *jws, const cjose_jwk_t *jwk, cj
     memset(jws->sig, 0, jws->sig_len);
 
     const BIGNUM *pr, *ps;
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
     ECDSA_SIG_get0(ecdsa_sig, &pr, &ps);
 #else
     pr = ecdsa_sig->r;
@@ -1007,7 +1007,7 @@ static bool _cjose_jws_verify_sig_ec(cjose_jws_t *jws, const cjose_jwk_t *jwk, c
     ECDSA_SIG *ecdsa_sig = ECDSA_SIG_new();
     int key_len = jws->sig_len / 2;
 
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
     BIGNUM *pr = BN_new(), *ps = BN_new();
     BN_bin2bn(jws->sig, key_len, pr);
     BN_bin2bn(jws->sig + key_len, key_len, ps);

--- a/src/util.c
+++ b/src/util.c
@@ -50,7 +50,7 @@ static void cjose_apply_allocs()
 {
     // set upstream
     json_set_alloc_funcs(cjose_get_alloc(), cjose_get_dealloc());
-#if (CJOSE_OPENSSL_11X)
+#if defined(CJOSE_OPENSSL_11X)
     CRYPTO_set_mem_functions(cjose_get_alloc3(), cjose_get_realloc3(), cjose_get_dealloc3());
 #else
     CRYPTO_set_mem_functions(cjose_get_alloc(), cjose_get_realloc(), cjose_get_dealloc());

